src/runtime/d3d12compute.cpp

```diff
@@ -0,0 +1,2765 @@
+#ifndef BITS_64
+    #pragma message "The Halide Direct3D 12 back-end is not yet supported on 32bit targets..."
+#endif
+
+#define HALIDE_D3D12_TRACE          (0)
+#define HALIDE_D3D12_DEBUG_LAYER    (0)
+#define HALIDE_D3D12_DEBUG_SHADERS  (0)
+#define HALIDE_D3D12_PROFILING      (0)
+#define HALIDE_D3D12_PIX            (0)
+#define HALIDE_D3D12_RENDERDOC      (0)
+
+#ifdef DEBUG_RUNTIME
+
+    #ifdef  HALIDE_D3D12_TRACE
```

Comment:
This works because the items were declared above. I would expect the code to either declare the macro to 0 or 1 based on DEBUG_RUNTIME or to undef it if it is defined and then unconditionally define it to 1.

Reply:
Often it helps me to just enable the d3d12 debug context and/or shader code debugging, without having to enable the call-tracing, that's why I declared them above. 
I'll remove the `#ifdef`/#endif` nesting, since indeed `#undef` should always work. Thanks for spotting that.